src/Integration.Vsix/Delegates/SonarAnalyzerLegacyConnectedWorkflow.cs

```diff
@@ -64,29 +62,19 @@ public SonarAnalyzerLegacyConnectedWorkflow(Workspace workspace, ISuppressionHan
             AppDomain.CurrentDomain.AssemblyLoad += OnAssemblyLoad;
         }
 
-        internal /* for testing purposes */ bool ShouldExecuteVsixAnalyzer(IAnalysisRunContext context)
+        internal /* for testing purposes */ bool ShouldIssueBeReported(SyntaxTree syntaxTree, Diagnostic diagnostic)
```

Comment:
Suggestion about behaviour on dispose: would it be neater to do an explicit check here for "_if (this.disposed) { return true; }_"? The comment about why the method might be called post-dispose could go here too.

It would make it clearer that this is an expected behaviour, and make it easy to see that the method is returning the expected value.

Reply:
Another alternative I thought of was to make the _shouldDiagnosticBeReportedFunc_ static, and in the _Dispose_ set it to a new lambda that returns the expected default. I'm not sure which I prefer - calling a method on a disposed object, or using a static field ;-)

Reply:
That's exactly what I ended up doing :)